README
NAME

    Pod::Spell::CommonMistakes - Catches common typos in POD

VERSION

      This document describes v1.001 of Pod::Spell::CommonMistakes - released October 31, 2014 as part of Pod-Spell-CommonMistakes.

SYNOPSIS

            #!/usr/bin/perl
            use strict; use warnings;
    
            use Pod::Spell::CommonMistakes qw( check_pod );
    
            my $file = $ARGV[0] || 'lib/Pod/Spell/CommonMistakes.pm';
            my $result = check_pod( $file );
            if ( keys %$result == 0 ) {
                    print "File passed tests!\n";
            } else {
                    print "File failed tests!\n";
                    foreach my $k ( keys %$result ) {
                            print " Found: '$k' - Possible spelling: '$result->{$k}'?\n";
                    }
            }

DESCRIPTION

    This module looks for any typos in your POD. It differs from Pod::Spell
    or Test::Spelling because it uses a custom wordlist and doesn't use the
    system spellchecker. The idea for this came from the
    http://wiki.debian.org/Teams/Lintian code in Debian, thanks!

    To use this, just pass it a filename that has POD in it and you'll get
    a hashref back. If the hashref is empty that means the checker found no
    misspelled words. If it contains keys, then the keys are the bad words
    and the values are the suggested spelling.

METHODS

 check_pod( $filename )

    This function is what you will usually run. It will run the spell
    checks against the POD in $filename. Warning: you would need to catch
    any exceptions thrown from this function!

    It returns a hashref of misspelled words and their suggested spelling.
    If the hash is empty then there is no errors in the POD.

 check_pod_case( $filename )

    This function behaves the same as "check_pod( $filename )" but it uses
    a "case" wordlist instead. The difference is that this wordlist will
    make sure you capitalize common terms properly. One example is:
    OpenLdap => OpenLDAP.

    NOTE: This does NOT run the same checks as "check_pod( $filename )"!
    You would need to use the "check_pod_all( $filename )" function.

 check_pod_all( $filename )

    This function behaves the same as "check_pod( $filename )" but it runs
    all the extra checks too. Currently it's just the case wordlist but
    others might be added in the future...

EXPORT

    You would need to manually get the function you want.

SEE ALSO

    Please see those modules/websites for more information related to this
    module.

      * Pod::Spell

      * Test::Pod::Spelling::CommonMistakes

SUPPORT

 Perldoc

    You can find documentation for this module with the perldoc command.

      perldoc Pod::Spell::CommonMistakes

 Websites

    The following websites have more information about this module, and may
    be of help to you. As always, in addition to those websites please use
    your favorite search engine to discover more resources.

      * MetaCPAN

      A modern, open-source CPAN search engine, useful to view POD in HTML
      format.

      http://metacpan.org/release/Pod-Spell-CommonMistakes

      * Search CPAN

      The default CPAN search engine, useful to view POD in HTML format.

      http://search.cpan.org/dist/Pod-Spell-CommonMistakes

      * RT: CPAN's Bug Tracker

      The RT ( Request Tracker ) website is the default bug/issue tracking
      system for CPAN.

      http://rt.cpan.org/NoAuth/Bugs.html?Dist=Pod-Spell-CommonMistakes

      * AnnoCPAN

      The AnnoCPAN is a website that allows community annotations of Perl
      module documentation.

      http://annocpan.org/dist/Pod-Spell-CommonMistakes

      * CPAN Ratings

      The CPAN Ratings is a website that allows community ratings and
      reviews of Perl modules.

      http://cpanratings.perl.org/d/Pod-Spell-CommonMistakes

      * CPAN Forum

      The CPAN Forum is a web forum for discussing Perl modules.

      http://cpanforum.com/dist/Pod-Spell-CommonMistakes

      * CPANTS

      The CPANTS is a website that analyzes the Kwalitee ( code metrics )
      of a distribution.

      http://cpants.cpanauthors.org/dist/overview/Pod-Spell-CommonMistakes

      * CPAN Testers

      The CPAN Testers is a network of smokers who run automated tests on
      uploaded CPAN distributions.

      http://www.cpantesters.org/distro/P/Pod-Spell-CommonMistakes

      * CPAN Testers Matrix

      The CPAN Testers Matrix is a website that provides a visual overview
      of the test results for a distribution on various Perls/platforms.

      http://matrix.cpantesters.org/?dist=Pod-Spell-CommonMistakes

      * CPAN Testers Dependencies

      The CPAN Testers Dependencies is a website that shows a chart of the
      test results of all dependencies for a distribution.

      http://deps.cpantesters.org/?module=Pod::Spell::CommonMistakes

 Email

    You can email the author of this module at APOCAL at cpan.org asking
    for help with any problems you have.

 Internet Relay Chat

    You can get live help by using IRC ( Internet Relay Chat ). If you
    don't know what IRC is, please read this excellent guide:
    http://en.wikipedia.org/wiki/Internet_Relay_Chat. Please be courteous
    and patient when talking to us, as we might be busy or sleeping! You
    can join those networks/channels and get help:

      * irc.perl.org

      You can connect to the server at 'irc.perl.org' and join this
      channel: #perl-help then talk to this person for help: Apocalypse.

      * irc.freenode.net

      You can connect to the server at 'irc.freenode.net' and join this
      channel: #perl then talk to this person for help: Apocal.

      * irc.efnet.org

      You can connect to the server at 'irc.efnet.org' and join this
      channel: #perl then talk to this person for help: Ap0cal.

 Bugs / Feature Requests

    Please report any bugs or feature requests by email to
    bug-pod-spell-commonmistakes at rt.cpan.org, or through the web
    interface at
    http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Pod-Spell-CommonMistakes
    . You will be automatically notified of any progress on the request by
    the system.

 Source Code

    The code is open to the world, and available for you to hack on. Please
    feel free to browse it and play with it, or whatever. If you want to
    contribute patches, please send me a diff or prod me to pull from your
    repository :)

    https://github.com/apocalypse/perl-pod-spell-commonmistakes

      git clone git://github.com/apocalypse/perl-pod-spell-commonmistakes.git

AUTHOR

    Apocalypse <APOCAL@cpan.org>

ACKNOWLEDGEMENTS

    Props goes out to jawnsy@irc for pointing out a spelling mistake in
    POE, which prompted me to write this.

    THANKS goes out to the Debian Lintian code, as it was a great starting
    place! http://wiki.debian.org/Teams/Lintian

COPYRIGHT AND LICENSE

    This software is copyright (c) 2014 by Apocalypse.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.

    The full text of the license can be found in the LICENSE file included
    with this distribution.

DISCLAIMER OF WARRANTY

    THERE IS NO WARRANTY FOR THE PROGRAM, TO THE EXTENT PERMITTED BY
    APPLICABLE LAW. EXCEPT WHEN OTHERWISE STATED IN WRITING THE COPYRIGHT
    HOLDERS AND/OR OTHER PARTIES PROVIDE THE PROGRAM "AS IS" WITHOUT
    WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING, BUT NOT
    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
    PARTICULAR PURPOSE. THE ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE
    OF THE PROGRAM IS WITH YOU. SHOULD THE PROGRAM PROVE DEFECTIVE, YOU
    ASSUME THE COST OF ALL NECESSARY SERVICING, REPAIR OR CORRECTION.

    IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
    WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MODIFIES AND/OR
    CONVEYS THE PROGRAM AS PERMITTED ABOVE, BE LIABLE TO YOU FOR DAMAGES,
    INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES
    ARISING OUT OF THE USE OR INABILITY TO USE THE PROGRAM (INCLUDING BUT
    NOT LIMITED TO LOSS OF DATA OR DATA BEING RENDERED INACCURATE OR LOSSES
    SUSTAINED BY YOU OR THIRD PARTIES OR A FAILURE OF THE PROGRAM TO
    OPERATE WITH ANY OTHER PROGRAMS), EVEN IF SUCH HOLDER OR OTHER PARTY
    HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.



Changes
Revision history for Pod-Spell-CommonMistakes

1.001 2014-10-31T19:09:06Z UTC
 - Updated the wordlist
 - Add support for Pod::Spelling as an adaptor, RT#80625 ( thanks
   RIBASUSHI! )

 [STATISTICS]
 - code churn: 8 files changed, 212 insertions(+), 17 deletions(-)

1.000 2011-02-21T23:25:42Z UTC
 - Converted to Dist::Zilla for the release process
 - Add some more common misspellings, thanks RT#61505
 - Fix a typo in check_pod_case which prevented it from working

0.01 0000-00-00T00:00:00Z UTC
 - initial release



Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  'build_requires' => {
    'File::Find' => '0',
    'File::Temp' => '0',
    'Module::Build' => '0.3601',
    'Test::More' => '0.88'
  },
  'configure_requires' => {
    'Module::Build' => '0.3601'
  },
  'dist_abstract' => 'Catches common typos in POD',
  'dist_author' => [
    'Apocalypse <APOCAL@cpan.org>'
  ],
  'dist_name' => 'Pod-Spell-CommonMistakes',
  'dist_version' => '1.000',
  'license' => 'perl',
  'module_name' => 'Pod::Spell::CommonMistakes',
  'recommends' => {},
  'recursive_test_files' => 1,
  'requires' => {
    'Exporter' => '0',
    'IO::Scalar' => '2.110',
    'Pod::Spell' => '1.01',
    'perl' => '5.008'
  },
  'script_files' => []
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


lib/Pod/Spelling/CommonMistakes.pm
#
# This file is part of Pod-Spell-CommonMistakes
#
# This software is copyright (c) 2014 by Apocalypse.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
use strict; use warnings;
package Pod::Spelling::CommonMistakes;
$Pod::Spelling::CommonMistakes::VERSION = '1.001';
our $AUTHORITY = 'cpan:APOCAL';

# ABSTRACT: Adaptor for Pod::Spelling to use CommonMistakes as a checker

use parent 'Pod::Spelling'; # Thanks :)

use Pod::Spell::CommonMistakes::WordList qw( _check_common _check_case );

# We override our base...
no warnings 'redefine';

sub _init {
	shift if not ref $_[0];
	my $self = shift;

	# Nothing magical to do :)

	return $self;
}

# Accepts one or more lines of text, returns a list mispelt words.
sub _spell_check_callback {
        my $self = shift;
	my @lines = grep { length $_ } split /\s+/, join( ' ', @_ );
	my $err = _check_common( \@lines );
	$err = { %$err, %{ _check_case( \@lines ) } };

        return keys %$err;
}

1;

__END__

=pod

=encoding UTF-8

=for :stopwords Apocalypse

=head1 NAME

Pod::Spelling::CommonMistakes - Adaptor for Pod::Spelling to use CommonMistakes as a checker

=head1 VERSION

  This document describes v1.001 of Pod::Spelling::CommonMistakes - released October 31, 2014 as part of Pod-Spell-CommonMistakes.

=head1 SYNOPSIS

	die "Don't use this module directly. Please use Pod::Spelling instead.";

=head1 DESCRIPTION

Adaptor for L<Pod::Spelling> as a backend checker.

=head1 SEE ALSO

Please see those modules/websites for more information related to this module.

=over 4

=item *

L<Pod::Spell::CommonMistakes|Pod::Spell::CommonMistakes>

=back

=head1 AUTHOR

Apocalypse <APOCAL@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by Apocalypse.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

The full text of the license can be found in the
F<LICENSE> file included with this distribution.

=head1 DISCLAIMER OF WARRANTY

THERE IS NO WARRANTY FOR THE PROGRAM, TO THE EXTENT PERMITTED BY
APPLICABLE LAW.  EXCEPT WHEN OTHERWISE STATED IN WRITING THE COPYRIGHT
HOLDERS AND/OR OTHER PARTIES PROVIDE THE PROGRAM "AS IS" WITHOUT WARRANTY
OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING, BUT NOT LIMITED TO,
THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
PURPOSE.  THE ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE OF THE PROGRAM
IS WITH YOU.  SHOULD THE PROGRAM PROVE DEFECTIVE, YOU ASSUME THE COST OF
ALL NECESSARY SERVICING, REPAIR OR CORRECTION.

IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MODIFIES AND/OR CONVEYS
THE PROGRAM AS PERMITTED ABOVE, BE LIABLE TO YOU FOR DAMAGES, INCLUDING ANY
GENERAL, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES ARISING OUT OF THE
USE OR INABILITY TO USE THE PROGRAM (INCLUDING BUT NOT LIMITED TO LOSS OF
DATA OR DATA BEING RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR THIRD
PARTIES OR A FAILURE OF THE PROGRAM TO OPERATE WITH ANY OTHER PROGRAMS),
EVEN IF SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE POSSIBILITY OF
SUCH DAMAGES.

=cut


Makefile.PL
#
# This file is part of Pod-Spell-CommonMistakes
#
# This software is copyright (c) 2014 by Apocalypse.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
use strict; use warnings;
# This Makefile.PL for Pod-Spell-CommonMistakes was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.014
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.31.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'ExtUtils::MakeMaker' => '0',
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Catches common typos in POD",
  "AUTHOR" => "Apocalypse <APOCAL\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "Pod-Spell-CommonMistakes",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Pod::Spell::CommonMistakes",
  "PREREQ_PM" => {
    "Exporter" => 0,
    "IO::Scalar" => "2.110",
    "Pod::Spell" => "1.01",
    "Pod::Spelling" => 0,
    "parent" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.88"
  },
  "VERSION" => "1.001",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "Exporter" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Handle" => 0,
  "IO::Scalar" => "2.110",
  "IPC::Open3" => 0,
  "Module::Build::Tiny" => "0.039",
  "Pod::Spell" => "1.01",
  "Pod::Spelling" => 0,
  "Test::More" => "0.88",
  "parent" => 0,
  "strict" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
